src/build/GhosttyResources.zig

```diff
@@ -16,16 +16,29 @@ pub fn init(b: *std.Build, cfg: *const Config) !GhosttyResources {
 
     // Terminfo
     terminfo: {
+        const mkdir_step = RunStep.create(b, "make share/terminfo directory");
+        switch (cfg.target.result.os.tag) {
+            // windows mkdir shouldn't need "-p"
+            .windows => mkdir_step.addArgs(&.{"mkdir"}),
+            else => mkdir_step.addArgs(&.{ "mkdir", "-p" }),
+        }
+        mkdir_step.addArg(b.fmt("{s}/share/terminfo", .{b.install_path}));
+        try steps.append(&mkdir_step.step);
+
         // Encode our terminfo
         var str = std.ArrayList(u8).init(b.allocator);
         defer str.deinit();
         try terminfo.ghostty.encode(str.writer());
 
         // Write it
         var wf = b.addWriteFiles();
-        const src_source = wf.add("share/terminfo/ghostty.terminfo", str.items);
-        const src_install = b.addInstallFile(src_source, "share/terminfo/ghostty.terminfo");
-        try steps.append(&src_install.step);
+        const source = wf.add("ghostty.terminfo", str.items);
+
+        if (cfg.emit_terminfo) {
+            const source_install = b.addInstallFile(source, "share/terminfo/ghostty.terminfo");
+            source_install.step.dependOn(&mkdir_step.step);
```

Comment:
I think only the `copy` step that uses `cp` should depend on `mkdir`. `addInstallFile` does create the directory but the problem is since we're using `cp -r` (for the reasons stated in the comment) the directory isn't made.